src/components/file-browser/FilesInDirectory.ts

```diff
@@ -63,80 +70,99 @@ const FilesInDirectory = (props: FilesInDirectoryProps) => {
       flex: '1 0 0'
     }
   }, [
-    h(FilesMenu, {
-      disabled: editDisabled,
-      disabledReason: editDisabledReason,
-      provider,
-      selectedFiles,
-      onDeleteFiles: () => {
-        setSelectedFiles({})
+    h(Dropzone, {
+      disabled: editDisabled || uploadState.active,
+      style: { display: 'flex', flexFlow: 'column nowrap', height: '100%' },
+      activeStyle: { backgroundColor: colors.accent(0.2), cursor: 'copy' },
+      multiple: true,
+      maxFiles: 0, // no limit on number of files
+      onDropAccepted: async files => {
+        await uploadFiles(files)
         reload()
-      },
-    }),
-
-    span({
-      ref: loadedAlertElementRef,
-      'aria-live': 'polite',
-      className: 'sr-only',
-      role: 'alert',
-    }),
-    status === 'Loading' && span({
-      'aria-live': 'assertive',
-      className: 'sr-only',
-      role: 'alert',
-    }, [`Loading files in ${directoryLabel}`]),
-    files.length > 0 && h(Fragment, [
-      h(FilesTable, {
-        'aria-label': `Files in ${directoryLabel}`,
-        files,
+      }
+      // @ts-expect-error
+    }, [({ openUploader }) => h(Fragment, [
+      h(FilesMenu, {
+        disabled: editDisabled,
+        disabledReason: editDisabledReason,
+        provider,
         selectedFiles,
-        setSelectedFiles,
-        onClickFile
+        onClickUpload: openUploader,
+        onDeleteFiles: () => {
+          setSelectedFiles({})
+          reload()
+        },
+      }),
+
+      span({
+        ref: loadedAlertElementRef,
+        'aria-live': 'polite',
+        className: 'sr-only',
+        role: 'alert',
       }),
-      div({
+      status === 'Loading' && span({
```

Comment:
Would above be a good place to use `LoadedState` instead of the string literal `'Loading'`?

Reply:
`useFilesInDirectory` does use LoadedState.

https://github.com/DataBiosphere/terra-ui/blob/2751494cc4ca3401e1a9d253a0ab2c5fd3cf1fc0/src/libs/ajax/incremental-response/useIncrementalResponse.ts#L9-L15